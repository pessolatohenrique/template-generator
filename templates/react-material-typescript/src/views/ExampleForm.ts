// Silence is golden
export function ExampleForm() {}
