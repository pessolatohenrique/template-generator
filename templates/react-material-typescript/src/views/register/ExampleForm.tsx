import React, { useEffect, useState } from "react";
import {
  Container,
  Card,
  CardContent,
  Typography,
  Grid,
  TextField,
  Button,
  Select,
  MenuItem,
  InputLabel,
  FormControl,
  FormHelperText,
  FormGroup,
  FormControlLabel,
  Switch,
} from "@mui/material";
import { useForm } from "react-hook-form";
import axios from "axios";

import { THEME_COLOR } from "../../constants/default_settings";
import { REQUIRED_MESSAGE } from "../../constants/messages";
import BreadcrumbsWrapper from "../../components/BreadcrumbsWrapper";
import { IExampleForm } from "../../interfaces/FormData";
import { IBook, IBookList } from "../../interfaces/Book";

function createData({ book, author, pages, genre }: IBook) {
  return { book, author, pages, genre };
}

const rows: IBookList = {
  items: [
    createData({
      book: "Jogos Vorazes",
      author: "Suzanne Collins",
      pages: 320,
      genre: "Trilogia",
    }),
    createData({
      book: "Harry Potter e a Pedra Filosofal",
      author: "J.K Rowling",
      pages: 220,
      genre: "Trilogia",
    }),
    createData({
      book: "Me poupe!",
      author: "Nathalia Arcuri",
      pages: 200,
      genre: "Finanças",
    }),
    createData({
      book: "Sherlock Homes",
      author: "Arthur Conan Doyle",
      pages: 500,
      genre: "Suspense",
    }),
    createData({
      book: "O mistério dos sete relógios",
      author: "Agatha Christie",
      pages: 300,
      genre: "Suspense",
    }),
    createData({
      book: "Harry Potter e o Enigma do Príncipe",
      author: "J.K Rowling",
      pages: 220,
      genre: "Trilogia",
    }),
  ],
};

function ExampleForm() {
  const {
    register,
    handleSubmit,
    formState: { errors },
    watch,
  } = useForm<IExampleForm>();
  const [books, setBooks] = useState<IBookList>({ items: [] });

  const onSubmit = (data: IExampleForm) => console.log(data);

  const watchAuthor = watch("author");

  useEffect(() => {
    async function searchCep() {
      const cep = await axios.get(
        `https://viacep.com.br/ws/${process.env.REACT_APP_CEP_EXAMPLE}/json/`
      );
      console.log("cep example", cep);
    }

    searchCep();

    const author = watch("author");
    const books = [...rows.items].filter((item) => item.author === author);
    setBooks({ items: books });
  }, [watchAuthor]);

  return (
    <>
      <Container fixed>
        <br />
        <Card>
          <CardContent>
            <BreadcrumbsWrapper
              parentLink={{ link: "/livros", label: "Livros" }}
              childrenLabel="Novo"
            />
            <Typography
              gutterBottom
              variant="h5"
              component="h1"
              color={THEME_COLOR}
            >
              Novo Livro
            </Typography>
            <form onSubmit={handleSubmit(onSubmit)}>
              <Grid container direction={"column"} spacing={2}>
                <Grid item>
                  <TextField
                    fullWidth
                    id="firstName"
                    label="First Name"
                    error={Boolean(errors.firstName)}
                    helperText={errors.firstName && REQUIRED_MESSAGE}
                    {...register("firstName", {
                      required: true,
                      maxLength: 20,
                    })}
                  />
                </Grid>
                <Grid item>
                  <TextField
                    fullWidth
                    id="lastName"
                    label="Lastname Name"
                    error={Boolean(errors.lastName)}
                    helperText={errors.lastName && REQUIRED_MESSAGE}
                    {...register("lastName", { required: true, maxLength: 20 })}
                  />
                </Grid>
                <Grid item>
                  <FormControl fullWidth>
                    <InputLabel id="author">Author</InputLabel>

                    <Select
                      labelId="author"
                      id="demo-simple-select-helper"
                      fullWidth
                      inputProps={{ "data-testid": "author" }}
                      error={Boolean(errors.author)}
                      {...register("author", {
                        required: true,
                      })}
                    >
                      <MenuItem value={"J.K Rowling"}>JK Rowling</MenuItem>
                      <MenuItem value={"Cortella"}>Cortella</MenuItem>
                      <MenuItem value={"Nathalia Arcuri"}>
                        Nathalia Arcuri
                      </MenuItem>
                    </Select>
                    {errors.author && (
                      <FormHelperText error>{REQUIRED_MESSAGE}</FormHelperText>
                    )}
                  </FormControl>
                </Grid>
                <Grid item>
                  <FormControl fullWidth>
                    <InputLabel id="book">Book</InputLabel>

                    <Select
                      labelId="book"
                      id="book"
                      fullWidth
                      inputProps={{ "data-testid": "book" }}
                      error={Boolean(errors.book)}
                      {...register("book", {
                        required: true,
                      })}
                    >
                      {books.items.map((item) => {
                        return (
                          <MenuItem key={item.book} value={item.book}>
                            {item.book}
                          </MenuItem>
                        );
                      })}
                    </Select>
                    {errors.book && (
                      <FormHelperText error>{REQUIRED_MESSAGE}</FormHelperText>
                    )}
                  </FormControl>
                </Grid>
                <Grid item>
                  <FormGroup>
                    <FormControlLabel
                      control={
                        <Switch defaultChecked {...register("active")} />
                      }
                      label="Ativo?"
                    />
                  </FormGroup>
                </Grid>
                <Grid item>
                  <Button variant="contained" type="submit">
                    Salvar
                  </Button>
                </Grid>
              </Grid>
            </form>
          </CardContent>
        </Card>
      </Container>
    </>
  );
}
export default ExampleForm;
