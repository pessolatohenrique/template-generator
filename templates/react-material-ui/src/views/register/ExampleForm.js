import React from "react";
import {
  Container,
  Card,
  CardContent,
  Typography,
  Grid,
  TextField,
  Button,
} from "@mui/material";
import { useForm } from "react-hook-form";

import { THEME_COLOR } from "../../constants/default_settings";
import BreadcrumbsWrapper from "../../components/BreadcrumbsWrapper";

function ExampleForm() {
  const {
    register,
    handleSubmit,
    formState: { errors },
  } = useForm();
  const onSubmit = (data) => console.log(data);

  return (
    <>
      <Container fixed>
        <br />
        <Card>
          <CardContent>
            <BreadcrumbsWrapper
              parentLink={{ link: "/livros", label: "Livros" }}
              childrenLabel="Novo"
            />
            <Typography
              gutterBottom
              variant="h5"
              component="h1"
              color={THEME_COLOR}
            >
              Novo Livro
            </Typography>
            <form onSubmit={handleSubmit(onSubmit)}>
              <Grid container direction={"column"} spacing={2}>
                <Grid item>
                  <TextField
                    fullWidth
                    id="firstName"
                    label="First Name"
                    error={Boolean(errors.firstName)}
                    helperText={
                      errors.firstName && "Campo de preenchimento obrigatório"
                    }
                    {...register("firstName", {
                      required: true,
                      maxLength: 20,
                    })}
                  />
                </Grid>
                <Grid item>
                  <TextField
                    fullWidth
                    id="lastName"
                    label="Lastname Name"
                    error={Boolean(errors.lastName)}
                    helperText={
                      errors.lastName && "Campo de preenchimento obrigatório"
                    }
                    {...register("lastName", { required: true, maxLength: 20 })}
                  />
                </Grid>
                <Grid item>
                  <Button variant="contained" type="submit">
                    Salvar
                  </Button>
                </Grid>
              </Grid>
            </form>
          </CardContent>
        </Card>
      </Container>
    </>
  );
}
export default ExampleForm;
