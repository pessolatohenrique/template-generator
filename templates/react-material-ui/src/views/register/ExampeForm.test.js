import { render, screen, fireEvent, waitFor } from "@testing-library/react";
import ExampleForm from "./ExampleForm";

describe("Example Form", () => {
  it("should generate snapshot", () => {
    const { container } = render(<ExampleForm />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should validate fields when in blank", async () => {
    render(<ExampleForm />);
    const submitBtn = screen.getByTestId("submit-button");
    fireEvent.click(submitBtn);

    waitFor(() => {
      expect(screen.getByTestId("firstName").toHaveClass("Mui-error"));
    });
  });

  it("should submit form", async () => {
    const { container } = render(<ExampleForm />);
    const submitBtn = screen.getByTestId("submit-button");
    const firstName = screen.getByTestId("firstName");
    const lastName = screen.getByTestId("lastName");

    fireEvent.input(firstName, { target: { value: "Henrique" } });
    fireEvent.input(lastName, { target: { value: "Henrique" } });
    fireEvent.click(submitBtn);

    waitFor(() => {
      expect(container.getElementsByClassName("Mui-error")).toHaveLength(0);
    });
  });
});
