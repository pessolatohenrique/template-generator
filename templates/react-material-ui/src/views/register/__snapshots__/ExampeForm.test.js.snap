// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example Form should generate snapshot 1`] = `
<div
  class="MuiContainer-root MuiContainer-maxWidthLg MuiContainer-fixed css-1ss2ei0-MuiContainer-root"
>
  <br />
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <nav
        aria-label="breadcrumb"
        class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
      >
        <ol
          class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
        >
          <li
            class="MuiBreadcrumbs-li"
          >
            <a
              class="MuiTypography-root MuiTypography-body2 MuiLink-root MuiLink-underlineHover css-t7xl82-MuiTypography-root-MuiLink-root"
              data-testid="bread-firstlevel"
              href="/"
            >
              Início
            </a>
          </li>
          <li
            aria-hidden="true"
            class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
          >
            ›
          </li>
          <li
            class="MuiBreadcrumbs-li"
          >
            <a
              class="MuiTypography-root MuiTypography-body2 MuiLink-root MuiLink-underlineHover css-t7xl82-MuiTypography-root-MuiLink-root"
              data-testid="bread-secondlevel"
              href="/livros"
            >
              Livros
            </a>
          </li>
          <li
            aria-hidden="true"
            class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
          >
            ›
          </li>
          <li
            class="MuiBreadcrumbs-li"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-1yr8029-MuiTypography-root"
              data-testid="bread-lastlevel"
            >
              Novo
            </p>
          </li>
        </ol>
      </nav>
      <h1
        class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-1y15g74-MuiTypography-root"
      >
        Novo Livro
      </h1>
      <form>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column css-1rl1og4-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="firstName"
                id="firstName-label"
              >
                First Name
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  data-testid="firstName"
                  id="firstName"
                  name="firstName"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-dhh0px"
                  >
                    <span>
                      First Name
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="lastName"
                id="lastName-label"
              >
                Lastname Name
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  data-testid="lastName"
                  id="lastName"
                  name="lastName"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-dhh0px"
                  >
                    <span>
                      Lastname Name
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <button
              class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-sghohy-MuiButtonBase-root-MuiButton-root"
              data-testid="submit-button"
              tabindex="0"
              type="submit"
            >
              Salvar
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
